pkg/client/s3client.go

```diff
@@ -153,18 +154,22 @@ func (c *S3Client) CheckBucketLocationConstraint(bucketName string, bucketLocati
 	return false, errors.New("bucket location constraint doesn't match")
 }
 
-func (c *S3Client) CheckIfObjectExists(bucketName, objectName string) bool {
+func (c *S3Client) CheckIfObjectExists(bucketName, objectName string) (bool, error) {
 	input := &s3.GetObjectInput{
 		Bucket: aws.String(bucketName),
 		Key:    aws.String(objectName),
 	}
 
 	_, err := c.S3Session.GetObject(input)
+
 	if err != nil {
-		klog.Errorf("ERROR: %v", err)
-		return false
+		if err.(awserr.Error).Code() == s3.ErrCodeNoSuchKey {
```

Comment:
There is a chance that the error itself may not be a awserr type, that time you will have an issue for the type casting, you can use following approach to check if it is really aws error first.

ref: 
https://github.com/IBM/ibm-cos-sdk-go/blob/c338dc6802b296727d4d194cc5d2818a17f98f9f/example/aws/request/handleServiceErrorCodes/handleServiceErrorCodes.go#L52:L60

Reply:
Done.